woocommerce/woocommerce-admin
# update return customer total to count in range

Fixes #4410 

The current calculation of the number of returning customers is the number of customers minus the number of new customers in the date range of the report. This calculates a reasonable total on a short range report (eg. week) but a misleading number on longer range reports (eg. last year).

Screenshots from my dev install using the `Last Year` filter

Current:
<img width="877" alt="Screen Shot 2020-05-22 at 10 10 33 AM" src="https://user-images.githubusercontent.com/343847/82672840-47bd7180-9c17-11ea-9bb5-b66fd192c9c4.png">

With this PR:
<img width="875" alt="Screen Shot 2020-05-22 at 10 11 06 AM" src="https://user-images.githubusercontent.com/343847/82672910-5f94f580-9c17-11ea-88b7-c070508d2b0c.png">

It will be more useful for this store owner to know that they have approximately 40% returning customers than 5% of the customers in this date range had previous orders.

This PR updates the returning customer count to be the count of unique returning customer IDs in the report date range.

### Detailed test instructions:

- Create an order using new billing information
- Check the orders report for the day to verify it shows as a `New` customer
- Create a second order using the same billing information
- Check the orders report 
  - The table shows one `New` and one `Returning` order
  - The summary should show `1 new customer` and `1 returning customer`

### Changelog Note:

Fix: Update returning customer total to include customers whose first order was within the report date range.
